portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -83,15 +101,55 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     } else {
       url = userService.createLoginURL("/");
     }
+
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    String uploadUrl = blobstoreService.createUploadUrl("/data");
     
-    ServerResponse resp = new ServerResponse(comments, userEmail, url);
+    ServerResponse resp = new ServerResponse(comments, userEmail, url, uploadUrl);
 
     response.setContentType("text/json;");
     response.getWriter().println(convertToJsonUsingGson(resp));
   }
 
-  private String convertToJsonUsingGson(ServerResponse list) {
+  private String convertToJsonUsingGson(ServerResponse resp) {
     Gson gson = new Gson();
-    return gson.toJson(list);
+    return gson.toJson(resp);
+  }
+
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get("image");
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // We could check the validity of the file here, e.g. to make sure it's an image file
+    // https://stackoverflow.com/q/10779564/873165
```

Comment:
We usually mark this as a TODO comment, e.g.: 
```
// TODO: We could check ...
```
This way it's easier to find these later on.

Reply:
Sorry, I copied that function from the blobstore example as it is, but I'll change it, thanks.